source/event_stream_rpc_server.c

```diff
@@ -922,13 +922,6 @@ static void s_route_message_by_type(
                     AWS_LS_EVENT_STREAM_RPC_SERVER,
                     "id=%p: stream_id is invalid because it's not sequentially increasing",
                     (void *)connection);
-
-                aws_raise_error(AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR);
```

Comment:
I believe this error is also rais-able by the client, so the same change should be applied there.

Reply:
The error is raise-able by the client, but the client doesn't have a condition for checking for stream_id non-sequentially increasing that I could see. I will double check though.